resources/hpxml-measures/BuildResidentialHPXML/resources/geometry.rb

```diff
@@ -105,8 +105,7 @@ def self.add_rim_joist(model, polygon, space, rim_joist_height, z)
       rim_joist_polygon = p
 
       # make space
-      rim_joist_space = OpenStudio::Model::Space::fromFloorPrint(rim_joist_polygon, rim_joist_height, model)
-      rim_joist_space = rim_joist_space.get
+      rim_joist_space = from_floor_print(rim_joist_polygon, rim_joist_height, model)
```

Comment:
So you verified that the `Space::fromFloorPrint` method is not deterministic? Would it make sense to update OS so that the code **is** deterministic rather than roll your own code? (Note: There have been [OS improvements in the past](https://github.com/NREL/OpenStudio/issues?q=is%3Aissue+is%3Aclosed+deterministic) to address non-deterministic results.) And what makes your code deterministic when the SDK code is not?

Reply:
`Space::fromFloorPrint` is deterministic. The problem is that if you went to retrieve all the surfaces it creates, the order of surfaces returned is not deterministic (i.e., if you wanted to then assign an index to each of them).

Reply:
I assume you always get back 1 ceiling, 1 floor, and multiple walls. So for the walls, can't you simply assign an index to those surfaces based on the order of edges in the `rim_joist_polygon`?

Reply:
Yeah, I thought about doing it that way. I guess it didn't necessarily seem simpler to me.